Assets/PoisonMushroom.cs

```diff
@@ -0,0 +1,102 @@
+﻿using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+
+public class PoisonMushroom : MonoBehaviour
+{
+    //Mushroom stats
+    public int health; //Mushroom health
+    public int damageNum; //amount of damage applied to Mushroom health
+    public float lifeTimer; //timer for destroying the Mushroom
+    private float generateTimer; //timer for generating the clouds
+    private BoxCollider collider;
+    private MeshRenderer mesh;
+
+    //Generating clouds using Particle System
+    private ParticleSystem cloud;
+    public float timeGenerate; //duration for generating the clouds
+
+    void Start()
+    {
+        collider = GetComponent<BoxCollider>();
+        mesh = GetComponent<MeshRenderer>();
+        cloud = GetComponent<ParticleSystem>();
+        generateTimer = timeGenerate;
+    }
+
+    void Update()
+    {
+        if (generateTimer > 0)
+        {
+            generateTimer -= Time.deltaTime;
+        }
+        else if (generateTimer == 0)
+        {
+            cloud.Stop();
+            cloud.Clear();
+        }
+        else if (generateTimer <= 0)
```

Comment:
this technically wouldn't cause any problems since ur doing an if else if else if statement but at the same time this last else if statement shouldn't check if its = 0 since that was handled by the previous one